### PR TITLE
feat(store): Save key_id as part of event data

### PR DESCRIPTION
**Pull request by macqueen**
refs GH-2709

also see https://github.com/getsentry/sentry/pull/6479#discussion_r147876369

### CODE REVIEW

File: src/sentry/web/api.py [hunk 1]
```diff
@@ -492,6 +492,7 @@ def process(self, request, project, key, auth, helper, data, **kwargs):
             helper.ensure_does_not_have_ip(data)
 
         # mutates data (strips a lot of context if not queued)
+        data['key_id'] = key.id
```
Comment (mattrobenolt) [inline src/sentry/web/api.py]: The only thing I'd really argue is that this should belong inside of the `ClientApiHelper` where other normalizations and mutations occur.

**Review by mattrobenolt — APPROVED** [review]
Also going to cc @alex-hofsteede for his context.

File: src/sentry/web/api.py (hunk 1 above)
Comment (mattrobenolt) [reply src/sentry/web/api.py]: I guess probably the best place seems to be `LazyData._decode`. You can pass along `key` to the constructor, then apply it in `_decode` similar to how `sdk` is applied.

**Review by mattrobenolt — CHANGES_REQUESTED** [review]

File: src/sentry/web/api.py (hunk 1 above)
Comment (alex-hofsteede) [reply src/sentry/web/api.py]: Agree this should probably be added before filters etc are applied. https://github.com/getsentry/sentry/blob/master/src/sentry/coreapi.py#L418 seems to be the other place where we do something similar.

Comment (alex-hofsteede) [reply src/sentry/web/api.py]: Of course if you could make it so that `sdk`, `project` and `key` were all set in the same place, that would be amazing

Comment (macqueen) [reply src/sentry/web/api.py]: 👍  moved both project and key to `LazyData._decode`

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/coreapi.py [hunk 2]
```diff
@@ -1085,6 +1080,12 @@ def _decode(self):
 
         data['sdk']['client_ip'] = self._client_ip
 
+        if 'project' not in data:
+            data['project'] = self._project.id
+
+        if 'key_id' not in data:
```
Comment (mattrobenolt) [inline src/sentry/coreapi.py]: This shouldn't be conditionally setting. We can't allow the client to send these values and override our fixed values. That means someone sending in their own key_id will be problematic.

**Review by mattrobenolt — CHANGES_REQUESTED** [review]

**Review by mattrobenolt — APPROVED** [review]
:unicorn:

**Pull request merged**